Create example module

robvanderleek opened the pull request:
Fixes #41 

Hi maintainers,
This is my first PR to your project in the spirit of Hacktoberfest 2017.
I've restructured the codebase into a Maven multi-module project with 'library' and 'sample' modules. Let me know if this is what you had in mind. I might have broken the Gradle build or the deploy phase.
Small question: is it necessary to have the license in all code files since there's already a LICENSE file?

MALPI commented:
Thanks a Lot for your contribution! I will Take a detailed look tomorrow! 
No its not necessary to have license Information in each File. I will File an issue for that.

MALPI commented on sample/src/main/java/org/zalando/failsafeactuator/FailsafeSampleApp.java:
```diff
@@ -0,0 +1,100 @@
+
+    }
+
+    public static void main(final String[] args) {
```
I would prefer having this in a separate class. Would you mind moving this?

robvanderleek replied:
I've broken up the sample application into a main, controller and banner part. What do you think of it now?

robvanderleek commented on .travis.yml:
```diff
@@ -8,7 +8,8 @@ before_install:
   - wget -O ~/codacy-coverage-reporter-assembly-latest.jar $(curl https://api.github.com/repos/codacy/codacy-coverage-reporter/releases/latest | jq -r .assets[0].browser_download_url)
 script:
- - mvn clean package
+  - mvn clean compile install -DskipTests -Dgpg.skip
```
Just `mvn test` works fine indeed, removed the first step.

robvanderleek commented on README.md:
````diff
@@ -89,6 +89,14 @@ The generated output will look like this:
+### Running the sample application
+
+```bash
+mvn compile install -Dgpg.skip
````
Yes, also dropped the first step.

MALPI commented:
Really good @robvanderleek! Just checked it out. Just two minor things.

The example URLs you print do have a static port configured, where the port is set in `application.properties`. If one changes this, the output is wrong. Could you either resolve it from `PropertyResolver` or replace it with `$baseUrl`?

Furthermore it's unclear to me what the `reliableWithDelay` endpoint should do? 

Last but not least, don't forget to add yourself as contributor to the `MAINTAINERS.MD`! 👍 

If you like the lib, leave it a star.

robvanderleek commented:
Thanks @MALPI
I've made the port number dynamic based on the environment variable `server.port`.
The `reliableWithDelay` endpoint has a circuit breaker that waits 5 seconds before it closes again. The `reliable` endpoint has a circuit breaker that closes immediately. I've modified the banner a bit, please let me know if this is still too confusing.
I've added myself as a contributor 😄 perhaps I can pick up another issue in this lib for Hacktoberfest 2017.

MALPI commented:
:+1:

MALPI commented:
@robvanderleek You could port the example to spring-boot-2 branch if you like. Feel free to open an issue.

mmodabber commented:
:+1:
